sql-plugin/src/main/spark332db/scala/com/nvidia/spark/rapids/shims/shimExpressions.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "332db"}
+{"spark": "341db"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids.shims
+
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.trees.TreePattern._
+
+trait ShimGetArrayStructFields extends ExtractValue {
+  override def nodePatternsInternal(): Seq[TreePattern] = Seq(EXTRACT_ARRAY_SUBFIELDS)
+}
+
+trait ShimGetArrayItem extends ExtractValue {
+  override def nodePatternsInternal(): Seq[TreePattern] = Seq(GET_ARRAY_ITEM)
+}
+
+trait ShimGetStructField extends ExtractValue {
+  override def nodePatternsInternal(): Seq[TreePattern] = Seq(GET_STRUCT_FIELD)
+}
+
+// Fallback to the default definition of `deterministic`
+trait GpuDeterministicFirstLastCollectShim extends Expression
```

Comment:
looks like a copy of https://github.com/NVIDIA/spark-rapids/blob/0c3efa006be3c069e7381590013b56868a901c8e/sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/SparkShims.scala/#L90-L103

let us add 330db tag in this file and delete from SparkShims.scala